### PR TITLE
Fix trakt exceptions when connection timed out

### PR DESCRIPTION
**Pull request by fernandog**
Avoid this:
```
AA
AAtraktException: HTTPSConnectionPool(host='api-v2launch.trakt.tv', port=443): Read timed out. (read timeout=5)
AA    raise traktException(e)
AA  File "/home/osmc/SickRage/lib/trakt/trakt.py", line 97, in traktRequest
AA    TraktEpisodeWatchlist = self.trakt_api.traktRequest("sync/watchlist/episodes")
AA  File "/home/osmc/SickRage/sickbeard/traktChecker.py", line 503, in _getEpisodeWatchlist
AA    if self._getEpisodeWatchlist():
AA  File "/home/osmc/SickRage/sickbeard/traktChecker.py", line 237, in syncWatchlist
AA    self.syncWatchlist()
AA  File "/home/osmc/SickRage/sickbeard/traktChecker.py", line 82, in run
```

### CODE REVIEW

File: lib/trakt/trakt.py [hunk 1]
```diff
@@ -92,18 +92,21 @@ def traktRequest(self, path, data=None, headers=None, url=None, method='GET', co
...
                 # Retry the request, cloudflare had a proxying issue
                 logger.log(u'Retrying trakt api request: %s' % path, logger.WARNING)
                 return self.traktRequest(path, data, headers, url, method)
             elif code is 401:
-                logger.log(u'Unauthorized. Please check your Trakt settings', logger.WARNING)
                 if self.traktToken(refresh=True, count=count):
                     return self.traktRequest(path, data, headers, url, method)
-                raise traktAuthException(e)
+                else:
+                    logger.log(u'Unauthorized. Please check your Trakt settings', logger.WARNING)
```
Comment (fernandog) [inline lib/trakt/trakt.py]: @miigotu is this ok? If not self.traktToken than we show the log message

File: lib/trakt/trakt.py [hunk 2]
```diff
@@ -112,7 +115,7 @@ def traktRequest(self, path, data=None, headers=None, url=None, method='GET', co
                 logger.log(u'Trakt error (404) the resource does not exist: %s' % url + path, logger.WARNING)
                 return {}
             else:
-                raise traktException(e)
+                logger.log(u'Could not connect to Trakt. Code error: {0}'.format(code), logger.ERROR)
```
Comment (miigotu) [inline lib/trakt/trakt.py]: return {}

File: lib/trakt/trakt.py [hunk 3]
```diff
@@ -92,18 +92,21 @@ def traktRequest(self, path, data=None, headers=None, url=None, method='GET', co
...
             code = getattr(e.response, 'status_code', None)
             if not code:
+                if 'timed out' in e:
+                    logger.log(u'Timeout connecting to Trakt. Try to increase timeout value in Trakt settings', logger.WARNING)                      
                 # This is pretty much a fatal error if there is no status_code
-                # It means there basically was no response at all
-                raise traktException(e)
+                # It means there basically was no response at all                    
+                else:
+                    logger.log(u'Could not connect to Trakt. Error: {0}'.format(e), logger.WARNING)
```
Comment (miigotu) [inline lib/trakt/trakt.py]: missing return {}

File: lib/trakt/trakt.py [hunk 4]
```diff
@@ -112,7 +115,8 @@ def traktRequest(self, path, data=None, headers=None, url=None, method='GET', co
                 logger.log(u'Trakt error (404) the resource does not exist: %s' % url + path, logger.WARNING)
                 return {}
             else:
-                raise traktException(e)
+                logger.log(u'Could not connect to Trakt. Code error: {0}'.format(code), logger.ERROR)
+                return {}
```
Comment (miigotu) [inline lib/trakt/trakt.py]: actually just move this return to the left 4 spaces.

File: lib/trakt/trakt.py [hunk 5]
```diff
@@ -112,7 +115,8 @@ def traktRequest(self, path, data=None, headers=None, url=None, method='GET', co
                 logger.log(u'Trakt error (404) the resource does not exist: %s' % url + path, logger.WARNING)
                 return {}
```
Comment (miigotu) [inline lib/trakt/trakt.py]: and you can remove this return also

**Comment by miigotu** [comment]
It needs to return {} after that if/else chain or the code will continue to run through when there was a problem with the request

**Pull request merged**